src/library/objects/ShowcaseExporter.js

```diff
@@ -0,0 +1,873 @@
+    ShowcaseExporter.prototype._drawEquipInfo = function (equip, ctx, x, y, rowWidth, rowHeight, fake) {
+        var startY = y;
+        var fontSize = 10;
+        ctx.font = fontSize + "pt \"Helvetica Neue\", Helvetica, Arial, sans-serif";
+        ctx.fillStyle = "#000";
+
+        var available = rowWidth - rowHeight * 3.5;
+
+        var name = this._splitText(equip.name, ctx, available);
+        y = this._drawEquipName(name, ctx, x + 30, y, rowHeight, fontSize, fake);
+
+        if (equip.name !== KC3Master.slotitem(equip.masterId).api_name) {
+            name = this._splitText(KC3Master.slotitem(equip.masterId).api_name, ctx, available, "");
+            y = this._drawEquipName(name, ctx, x + 30, y, fontSize, fontSize, fake);
+        }
+        y = y < startY + 30 ? startY + 30 : y;
+
+        available = rowWidth - rowHeight * 3.5;
+
+        y += 5;
+        var statsY = this._drawEquipStats(equip, ctx, x + rowHeight, y, available, fake);
+        y = Math.max(statsY, y);
+        if (statsY > 0)
+            y += 10;
+        return y;
+    };
+
+    ShowcaseExporter.prototype._drawEquipName = function (nameLines, ctx, x, y, rowHeight, fontSize, fake) {
+        if (nameLines.length === 1) {
+            if (!fake)
+                ctx.fillText(nameLines[0], x, y + fontSize);
+            y += fontSize + 5;
+        } else {
+            for (var i = 0; i < nameLines.length; i++) {
+                if (!fake)
+                    ctx.fillText(nameLines[i], x, y + fontSize);
+                y += fontSize + 5;
+            }
+        }
+        return y;
+    };
+
+    ShowcaseExporter.prototype._splitText = function (text, ctx, maxWidth, splitter) {
+        if (typeof splitter === "undefined")
+            splitter = " ";
+
+        if (text.indexOf(splitter) === -1)
+            splitter = "";
+
+        var rows = [];
+        var words = text.split(splitter);
+
+        //@TODO what if single word will be so long so it goes outside maxWidth?
+        while (words.length > 0) {
+            var line = [];
+            var next = words.shift();
+            while (ctx.measureText((line.join(splitter) + splitter + next).trim()).width < maxWidth && next !== "") {
+                line.push(next);
+                if (words.length > 0)
+                    next = words.shift();
+                else
+                    next = "";
+            }
+            rows.push(line.join(splitter));
+            if (next !== "") {
+                words.unshift(next);
+            }
+        }
+
+        return rows;
+    };
+
+    ShowcaseExporter.prototype._drawEquipStats = function (equip, ctx, x, y, available, fake) {
+        var fontSize = 10;
+        ctx.font = fontSize + "pt \"Helvetica Neue\", Helvetica, Arial, sans-serif";
+        ctx.fillStyle = "#000";
+
+        var startX = x;
+        var img;
+        var noStats = true;
+
+        function drawStat(img) {
+            if (typeof equip[i] !== "undefined" && equip[i] > 0) {
```

Comment:
should be `equip[i] !== 0` because some equipment like those bulge hold negative stat.